orchagent/routeorch.cpp

```diff
@@ -1027,8 +1163,97 @@ bool RouteOrch::addRoute(sai_object_id_t vrf_id, const IpPrefix &ipPrefix, const
         route_attr.value.oid = next_hop_id;
 
         /* Default SAI_ROUTE_ATTR_PACKET_ACTION is SAI_PACKET_ACTION_FORWARD */
-        sai_status_t status = sai_route_api->create_route_entry(&route_entry, 1, &route_attr);
-        if (status != SAI_STATUS_SUCCESS)
+        object_statuses.emplace_back();
+        sai_status_t status = gRouteBulker.create_entry(&object_statuses.back(), &route_entry, 1, &route_attr);
+        if (status == SAI_STATUS_ITEM_ALREADY_EXISTS)
+        {
+            SWSS_LOG_ERROR("Failed to create route %s with next hop(s) %s: already exists in bulker",
+                    ipPrefix.to_string().c_str(), nextHops.to_string().c_str());
+            return false;
+        }
+    }
+    else
+    {
+        /* Set the packet action to forward when there was no next hop (dropped) */
+        if (it_route->second.getSize() == 0)
+        {
+            route_attr.id = SAI_ROUTE_ENTRY_ATTR_PACKET_ACTION;
+            route_attr.value.s32 = SAI_PACKET_ACTION_FORWARD;
+
+            object_statuses.emplace_back();
+            gRouteBulker.set_entry_attribute(&object_statuses.back(), &route_entry, &route_attr);
```

Comment:
Is there supposed to be an `emplace_back` in here somewhere similar to the other if branch?

Reply:
Seems I already have.

---
In reply to: [417495448](https://github.com/Azure/sonic-swss/pull/1238#discussion_r417495448) [](ancestors = 417495448)